### PR TITLE
📖 clarify backport-policy

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What this PR does / why we need it**:
As per office hours discussing clarifying backport policy for doc improvements, CI signal improvements, test framework improvements

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -78,7 +78,12 @@ The test module, clusterctl, and experiments do not provide any backward compati
 #### Backporting a patch
 
 We only accept backports of critical bugs, security issues, or bugs without easy workarounds, any
-backport MUST not be breaking for either API or behavioral changes.
+backport MUST not be breaking for either API or behavioral changes. In order to improve user/developer experience
+mantainers can choose to backport:
+- Doc improvements
+- Improvements to CI singal
```
Comment (sbueringer) [inline CONTRIBUTING.md:84]:
```suggestion
- Improvements to CI signal
```
nit

otherwise lgtm

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -78,7 +78,12 @@ The test module, clusterctl, and experiments do not provide any backward compati
 #### Backporting a patch
 
 We only accept backports of critical bugs, security issues, or bugs without easy workarounds, any
-backport MUST not be breaking for either API or behavioral changes.
+backport MUST not be breaking for either API or behavioral changes. In order to improve user/developer experience
+mantainers can choose to backport:
```
Comment (furkatgofurov7) [inline CONTRIBUTING.md:82]:
```suggestion
maintainers can choose to backport:
```

**Review by furkatgofurov7 — COMMENTED** [review]
small nit

**Comment by sbueringer** [comment]
/lgtm

I'll bring the topic up in the office hours today as well

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -78,7 +78,13 @@ The test module, clusterctl, and experiments do not provide any backward compati
...
 
 We only accept backports of critical bugs, security issues, or bugs without easy workarounds, any
-backport MUST not be breaking for either API or behavioral changes.
+backport MUST not be breaking for either API or behavioral changes. In order to improve user/developer experience
+maintainers can choose to backport:
+- Doc improvements
+- Improvements to CI signal
+- Improvements to the test framework (enabling improvements to provider's test signal)
+- Cert-manager bump (to avoid having branch using cert-manager versions out of support, when possible)
+
```
Comment (randomvariable) [inline CONTRIBUTING.md:87]: We probably need something here about changes required to support patch versions of Kubernetes that are supposed to be supported by the minor release. These will mostly be in the domain of kubeadmcontrolplane, but might include changes to the etcd clients, coredns libraries etc...

Comment (sbueringer) [reply CONTRIBUTING.md:87]: Good point (s/patch versions/minor versions/)

**Review by ykakarap — COMMENTED** [review]
/lgtm

**Comment by fabriziopandini** [comment]
Added a note for backporting changes required to support new Kubernetes versions, if possible

**Review by killianmuldoon — APPROVED** [review]
/lgtm

**Comment by fabriziopandini** [comment]
@CecileRobertMichon @enxebre @vincepri kindly ping

**Comment by CecileRobertMichon** [comment]
/lgtm
/approve
/hold 

Hold to give @enxebre @vincepri a chance to review. Feel free to unhold when ready to merge.

**Review by vincepri — APPROVED** [review]

**Comment by fabriziopandini** [comment]
/hold cancel

we got +1 from Vince and we are basically documenting what we are already doing since some time now

**Pull request merged**